internal/rootcoord/import_manager.go

```diff
@@ -205,15 +221,19 @@ func (m *importManager) sendOutTasks(ctx context.Context) error {
 		// Add new working dataNode to busyNodes.
 		m.busyNodes[resp.GetDatanodeId()] = true
 
-		func() {
+		err = func() error {
 			m.workingLock.Lock()
 			defer m.workingLock.Unlock()
 
 			log.Debug("import task added as working task", zap.Int64("task ID", it.TaskId))
 			task.State.StateCode = commonpb.ImportState_ImportPending
 			m.workingTasks[task.GetId()] = task
-			m.updateImportTaskStore(task)
+			err := m.updateImportTaskStore(task)
+			return err
 		}()
+		if err != nil {
+			return err
```

Comment:
no need return here.
the task has been sent by callImportService, that means a datanode is going to execute the task. the task ought to be put into working list.

Reply:
We need to first do Etcd update and then put the task in working list. We can then return on Etcd errors.

Reply:
like this?
```
		err = func() error {
			m.workingLock.Lock()
			defer m.workingLock.Unlock()

			log.Debug("import task added as working task", zap.Int64("task ID", it.TaskId))
			task.State.StateCode = commonpb.ImportState_ImportPending
			m.workingTasks[task.GetId()] = task
			err := m.updateImportTaskStore(task)
			return err
		}()
		// Erase this task from head of pending list.
		m.pendingTasks = append(m.pendingTasks[:0], m.pendingTasks[1:]...)

		if err != nil {
			return err
		}
```